test/unit_tests/unittest_scanner_configuration.cpp

```diff
@@ -46,71 +95,70 @@ class ScannerConfigurationTest : public testing::Test
 
 TEST_F(ScannerConfigurationTest, testConstructorSuccess)
 {
-  EXPECT_NO_THROW(ScannerConfiguration sc(
-      host_ip_, host_udp_port_data_, host_udp_port_control_, client_ip_, start_angle_, end_angle_));
+  EXPECT_NO_THROW(ScannerConfigurationBuilder().build());
 }
 
 TEST_F(ScannerConfigurationTest, testConstructorInvalidHostIp)
 {
-  host_ip_ = INVALID_IP;
-
-  EXPECT_THROW(ScannerConfiguration sc(
-                   host_ip_, host_udp_port_data_, host_udp_port_control_, client_ip_, start_angle_, end_angle_),
-               std::invalid_argument);
+  EXPECT_THROW(ScannerConfigurationBuilder().setHostIp(INVALID_IP).build(), std::invalid_argument);
 }
 
 TEST_F(ScannerConfigurationTest, testConstructorInvalidClientIp)
 {
-  client_ip_ = INVALID_IP;
-
-  EXPECT_THROW(ScannerConfiguration sc(
-                   host_ip_, host_udp_port_data_, host_udp_port_control_, client_ip_, start_angle_, end_angle_),
-               std::invalid_argument);
+  EXPECT_THROW(ScannerConfigurationBuilder().setClientIp(INVALID_IP).build(), std::invalid_argument);
 }
 
-TEST_F(ScannerConfigurationTest, testConstructorInvalidDataPort)
+TEST_F(ScannerConfigurationTest, testConstructorDataPortTooSmall)
 {
-  host_udp_port_data_ = std::numeric_limits<uint16_t>::min() - 1;
+  EXPECT_THROW(ScannerConfigurationBuilder().setHostUdpDataPort(-1).build(), std::invalid_argument);
+}
 
-  EXPECT_THROW(ScannerConfiguration sc(
-                   host_ip_, host_udp_port_data_, host_udp_port_control_, client_ip_, start_angle_, end_angle_),
+TEST_F(ScannerConfigurationTest, testConstructorDataPortTooLarge)
+{
+  EXPECT_THROW(ScannerConfigurationBuilder().setHostUdpDataPort(MAXIMAL_PORT_NUMBER + 1).build(),
                std::invalid_argument);
+}
 
-  host_udp_port_data_ = MAXIMAL_PORT_NUMBER + 1;
-
-  EXPECT_THROW(ScannerConfiguration sc(
-                   host_ip_, host_udp_port_data_, host_udp_port_control_, client_ip_, start_angle_, end_angle_),
+TEST_F(ScannerConfigurationTest, testConstructorControlPortTooSmall)
+{
+  EXPECT_THROW(ScannerConfigurationBuilder().setHostUdpControlPort(std::numeric_limits<uint16_t>::min() - 1).build(),
```

Comment:
Shouldn't this case be analogue to `testConstructorDataPortTooSmall`? There you use `setHostUdpDataPort(-1)`. Consider introducing a variable `MINIMAL_PORT_NUMBER`.